### PR TITLE
Fix process of gettign the active monitor

### PR DESCRIPTION
**Pull request by mcz**
- Get the window with keyboard focus using XGetInputFocus() instead of
  _NET_ACTIVE_WINDOW
- Handle edge cases of keyboard/mouse focus being on different X screen
- Set sane default in case the active monitor can't be determined
  (fixing issue #762)

### CODE REVIEW

**Comment by mcz** [comment]
This doesn't fix the general confusion between screens and monitors. Xrandr and Xinerama allow for multiple monitor to show parts of the same screen. Most references to "screens" in screen.c should be talking about monitors instead. It would be possible to handle multiple screens, but I don't think it would be worth it since most people don't use that part of X11 nowadays.

File: src/x11/screen.c [hunk 1]
```diff
@@ -327,19 +324,25 @@ const struct screen_info *get_active_screen(void)
...
+                                 * the screen the pointer is on, force following
+                                 * the mouse.
                                  */
                                 force_follow_mouse = true;
                         } else {
                                 Window child_return;
-                                XTranslateCoordinates(xctx.dpy, focused, root,
-                                                0, 0, &x, &y, &child_return);
+                                /*
+                                 * The window with input focus might me on a
```
Comment (fwSmit) [inline src/x11/screen.c:338]: me → be

File: src/x11/screen.c [hunk 2]
```diff
@@ -386,32 +389,13 @@ const struct screen_info *get_active_screen(void)
...
-                focused = *(Window *)prop_return;
-                XFree(prop_return);
-        }
+        Window focused;
+        int ignored;
+
+        XGetInputFocus(xctx.dpy, &focused, &ignored);
 
+        if (focused == None || focused == PointerRoot)
+                focused = 0;
```
Comment (fwSmit) [inline src/x11/screen.c:398]: None is a typedef for 0L, so you can leave that out of the if statement. I would also set focused to None for clarity.  Or are there cases where None != 0L?

Comment (tsipinakis) [reply src/x11/screen.c:398]: The typedef might not change but it's always best to do what the library devs expect you to in my experience :). It's not worth it for a 5 character reduction.

**Comment by fwSmit** [comment]
> This doesn't fix the general confusion between screens and monitors. Xrandr and Xinerama allow for multiple monitor to show parts of the same screen. Most references to "screens" in screen.c should be talking about monitors instead. It would be possible to handle multiple screens, but I don't think it would be worth it since most people don't use that part of X11 nowadays.

Yeah dunst's X11 implementation isn't the cleanest. Your PR defnitely helps.

As a side note, I'm wondering what do you need two X111 screens for?

**Comment by mcz** [comment]
> I'm wondering what do you need two X111 screens for?
It's just the most convenient way to handle my setup. I have two monitors with my WM (openbox) only running on the main one. The secondary monitor has only suckless's tabbed on it, with a small program listening for MapRequest events and adding the appropriate windows as tabs. It also monitors the number of active tabs, and if there are none, the monitor is turned of with xrandr. This means I only have two monitors when I need them.

File: src/x11/screen.c (hunk 2 above)
Comment (fwSmit) [reply src/x11/screen.c:398]: What do you mean? Are you for or against changing it to `focused = None`. I don't think they accidentally typedef'd None to 0.

Comment (mcz) [reply src/x11/screen.c:398]: None is typedef'd to 0L in Xorg, but that isn't guaranteed by the X11 spec. There might be a X11 implementation where None = 1L.

Comment (fwSmit) [reply src/x11/screen.c:398]: Allright then this is fine by me.

**Comment by fwSmit** [comment]
Interesting setup. If you squash the commits in one it looks good to me

**Review by tsipinakis — APPROVED** [review]
LGTM! Thanks for the PR! (and thanks @fwSmit for the review :) )

**Pull request merged**